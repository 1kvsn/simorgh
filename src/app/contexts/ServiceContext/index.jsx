import React from 'react';
import { node, string } from 'prop-types';
import services from '#lib/config/services/loadableConfig';
import variantPropType from '../../models/propTypes/variants';
import { getVariant } from '#lib/utilities/variantHandler';
import getLangOverride from '#lib/utilities/langHandler';

export const ServiceContext = React.createContext({});

export const ServiceContextProvider = ({
  children,
  service,
  variant,
  pageLang,
}) => {
  const LoadableContextProvider = services[service];

  if (!LoadableContextProvider) {
    return null;
  }

  return (
    <LoadableContextProvider
      Context={ServiceContext}
      dataKey={getLangOverride(pageLang) || getVariant({ service, variant })}
    >
      {children}
    </LoadableContextProvider>
  );
};

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string,
  variant: variantPropType,
  pageLang: string,
};

ServiceContextProvider.defaultProps = {
  service: 'default',
  variant: 'default',
  pageLang: null,
};
