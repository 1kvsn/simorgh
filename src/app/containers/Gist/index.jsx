import React, { useContext } from 'react';
import pathOr from 'ramda/src/pathOr';
import styled from '@emotion/styled';
import {
  GEL_SPACING_HLF,
  GEL_SPACING,
  GEL_SPACING_DBL,
  GEL_SPACING_QUAD,
} from '@bbc/gel-foundations/spacings';
import { C_METAL, C_POSTBOX, C_WHITE } from '@bbc/psammead-styles/colours';
import { getSansRegular, getSansBold } from '@bbc/psammead-styles/font-styles';
import { getDoublePica, getGreatPrimer } from '@bbc/gel-foundations/typography';
import {
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
} from '@bbc/gel-foundations/breakpoints';
import { string, arrayOf, shape } from 'prop-types';
import { ServiceContext } from '#contexts/ServiceContext';
import Blocks from '#containers/Blocks';
import UnorderedList from '../BulletedList';
import Text from '#containers/CpsText';
import { GridItemMedium } from '#app/components/Grid';

const GistWrapper = styled.div`
  ${({ service }) => getSansRegular(service)}
  ${({ script }) => getGreatPrimer(script)}
  color: ${C_METAL};
  border-top: ${GEL_SPACING_HLF} solid ${C_POSTBOX};
  background-color: ${C_WHITE};
  padding-top: ${GEL_SPACING_QUAD};
  padding-bottom: ${GEL_SPACING};
  margin-bottom: ${GEL_SPACING_QUAD};
  ${({ dir }) =>
    dir === 'ltr'
      ? `padding-left: ${GEL_SPACING_DBL};`
      : `padding-right: ${GEL_SPACING_DBL};`}
  ul {
    ${({ dir }) =>
      dir === 'ltr'
        ? `padding-left: ${GEL_SPACING};`
        : `padding-right: ${GEL_SPACING};`}
  }
  ul > li {
    margin-bottom: 10px;
    ${({ dir }) =>
      dir === 'ltr'
        ? `padding-left: ${GEL_SPACING};`
        : `padding-right: ${GEL_SPACING};`}
  }

  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    ul {
      ${({ dir }) => (dir === 'ltr' ? `padding-left: 0;` : `padding-right: 0;`)}
    }
  }

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    ${({ dir }) =>
      dir === 'ltr'
        ? `padding-left: ${GEL_SPACING_QUAD};`
        : `padding-right: ${GEL_SPACING_QUAD};`}
    ul > li {
      ${({ dir }) =>
        dir === 'ltr'
          ? `padding-left: ${GEL_SPACING_DBL};`
          : `padding-right: ${GEL_SPACING_DBL};`}
    }
  }

  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    ul {
      ${({ dir }) =>
        dir === 'ltr'
          ? `padding-left: ${GEL_SPACING_DBL};`
          : `padding-right: ${GEL_SPACING_DBL};`}
    }
  }
`;

const GistIntroduction = styled.strong`
  ${({ service }) => getSansBold(service)}
  ${({ script }) => getDoublePica(script)}
  display: inline-block;
  padding-bottom: ${GEL_SPACING_DBL};
`;

const componentsToRender = {
  text: props => (
    <Text
      {...props}
      componentsToRender={{
        unorderedList: innerProps => (
          <UnorderedList
            {...innerProps}
            bulletPointShape="square"
            bulletPointColour={C_METAL}
          />
        ),
      }}
    />
  ),
};

const Gist = ({ blocks }) => {
  const { service, script, dir, translations } = useContext(ServiceContext);
  const gistTitle = pathOr('At a glance', ['gist'], translations);
  return (
    <GridItemMedium>
      <GistWrapper service={service} script={script} dir={dir}>
        <GistIntroduction service={service} script={script}>
          {gistTitle}
        </GistIntroduction>
        <Blocks blocks={blocks} componentsToRender={componentsToRender} />
      </GistWrapper>
    </GridItemMedium>
  );
};

Gist.propTypes = {
  blocks: arrayOf(
    shape({
      type: string.isRequired,
      id: string.isRequired,
    }),
  ).isRequired,
};

export default Gist;
