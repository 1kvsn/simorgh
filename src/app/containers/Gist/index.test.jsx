import React from 'react';
import { render } from '@testing-library/react';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import Gist from '.';
import fixtureData, { fixtureDataOneItem } from './fixtures';

// eslint-disable-next-line react/prop-types
const GistWithContext = ({ gistData = fixtureData }) => (
  <ServiceContextProvider dir="ltr" service="news" lang="en-GB">
    <Gist gistData={gistData} />
  </ServiceContextProvider>
);

describe('Gist', () => {
  it('should render the gist heading', () => {
    const { getByText } = render(<GistWithContext />);
    const gistCaption = getByText('At a glance');

    expect(gistCaption).toBeInTheDocument();
  });

  it('should render an unordered list if there is more than one item', () => {
    const { container } = render(<GistWithContext />);

    expect(container.querySelectorAll('li').length).toEqual(2);
  });

  it.skip('should render the bullet point as a paragraph if there is only one item', () => {
    const { queryByRole } = render(
      <GistWithContext gistData={fixtureDataOneItem} />,
    );

    expect(queryByRole('list')).not.toBeInTheDocument();
    expect(queryByRole('listitem')).not.toBeInTheDocument();
  });
});
