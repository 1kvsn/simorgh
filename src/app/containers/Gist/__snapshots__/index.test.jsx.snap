// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gist should render an image with alt text 1`] = `
.emotion-0 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #6E6E73;
  border-top: 0.25rem solid #B80000;
  background-color: #FFFFFF;
  padding-top: 2rem;
  padding-bottom: 2rem;
  padding-left: 1rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-0 ul {
  padding-left: 0.5rem;
}

.emotion-0 ul>li {
  margin-bottom: 10px;
  padding-left: 0.5rem;
}

@media (min-width: 25rem) {
  .emotion-0 ul {
    padding-left: 0;
  }
}

@media (min-width: 63rem) {
  .emotion-0 {
    padding-left: 2rem;
  }

  .emotion-0 ul {
    padding-left: 1rem;
  }

  .emotion-0 ul>li {
    padding-left: 1rem;
  }
}

.emotion-2 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 1.125rem;
  line-height: 1.375rem;
  display: inline-block;
  padding-bottom: 1rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-2 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-2 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (max-width: 14.9375rem) {
  .emotion-4 {
    padding: 0 0.5rem;
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-4 {
    padding: 0 0.5rem;
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-4 {
    padding: 0 1rem;
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-4 {
    padding: 0 1rem;
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-4 {
    margin-left: 20%;
  }
}

@media (min-width: 80rem) {
  .emotion-4 {
    margin-left: 40%;
  }
}

@supports (display: grid) {
  .emotion-4 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      padding: 0 0.5rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      padding: 0 0.5rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-4 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      padding: 0 1rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      padding: 0 1rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-4 {
      grid-template-columns: repeat(10, 1fr);
      grid-column-end: span 10;
      grid-column-start: 5;
    }
  }
}

.emotion-6 {
  margin-bottom: 1.5rem;
}

.emotion-8 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  margin-top: 0;
  list-style-type: none;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-8 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-8 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

.emotion-8>li {
  position: relative;
}

.emotion-8>li::before {
  top: 0.5rem;
  content: ' ';
  position: absolute;
  border-width: 1rem;
  border: 0.1875rem solid #6E6E73;
  background-color: #6E6E73;
  border-radius: 0;
  left: -1rem;
}

.emotion-10 {
  margin-bottom: 1rem;
}

.emotion-14 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: italic;
  font-family: inherit;
  font-weight: inherit;
}

.emotion-18 {
  color: #222222;
  border-bottom: 1px solid #B80000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-18:visited {
  color: #6E6E73;
  border-bottom: 1px solid #6E6E73;
}

.emotion-18:focus,
.emotion-18:hover {
  border-bottom: 2px solid #B80000;
  color: #B80000;
}

<div>
  <div
    class="emotion-0 emotion-1"
    dir="ltr"
  >
    <strong
      class="emotion-2 emotion-3"
    >
      At a glance
    </strong>
    <div
      class="emotion-4 emotion-5"
      dir="ltr"
    >
      <div
        class="emotion-6 emotion-7"
      >
        <ul
          class="emotion-8 emotion-9"
          dir="ltr"
          role="list"
        >
          <li
            class="emotion-10 emotion-11"
            role="listitem"
          >
            Here is a summary of the article
          </li>
          <li
            class="emotion-10 emotion-11"
            role="listitem"
          >
            This component supports 
            <b>
              bold,
            </b>
             
            <i
              class="emotion-14 emotion-15"
            >
              italic,
            </i>
             
            <i
              class="emotion-14 emotion-15"
            >
              <b>
                bold-italic, 
              </b>
            </i>
            <a
              aria-label="links,, external"
              class="emotion-18 emotion-19"
              href="https://www.google.com"
            >
              links,
            </a>
             
            <a
              aria-label="and bold-italic-links,, external"
              class="emotion-18 emotion-19"
              href="https://www.google.com"
            >
              <i
                class="emotion-14 emotion-15"
              >
                <b>
                  and bold-italic-links,
                </b>
              </i>
            </a>
          </li>
          <li
            class="emotion-10 emotion-11"
            role="listitem"
          >
            It can be as long as editorial desires
          </li>
          <li
            class="emotion-10 emotion-11"
            role="listitem"
          >
            It does not support nested lists
          </li>
          <li
            class="emotion-10 emotion-11"
            role="listitem"
          >
            If a sentence is added that is really, really, really long, the text will wrap onto the next line, which is hopefully the behaviour of this sentence right now!
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Gist should render an image with alt text 2`] = `
.emotion-0 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #6E6E73;
  border-top: 0.25rem solid #B80000;
  background-color: #FFFFFF;
  padding-top: 2rem;
  padding-bottom: 2rem;
  padding-left: 1rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-0 ul {
  padding-left: 0.5rem;
}

.emotion-0 ul>li {
  margin-bottom: 10px;
  padding-left: 0.5rem;
}

@media (min-width: 25rem) {
  .emotion-0 ul {
    padding-left: 0;
  }
}

@media (min-width: 63rem) {
  .emotion-0 {
    padding-left: 2rem;
  }

  .emotion-0 ul {
    padding-left: 1rem;
  }

  .emotion-0 ul>li {
    padding-left: 1rem;
  }
}

.emotion-2 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 1.125rem;
  line-height: 1.375rem;
  display: inline-block;
  padding-bottom: 1rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-2 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-2 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (max-width: 14.9375rem) {
  .emotion-4 {
    padding: 0 0.5rem;
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-4 {
    padding: 0 0.5rem;
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-4 {
    padding: 0 1rem;
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-4 {
    padding: 0 1rem;
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-4 {
    margin-left: 20%;
  }
}

@media (min-width: 80rem) {
  .emotion-4 {
    margin-left: 40%;
  }
}

@supports (display: grid) {
  .emotion-4 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      padding: 0 0.5rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      padding: 0 0.5rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-4 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      padding: 0 1rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      padding: 0 1rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-4 {
      grid-template-columns: repeat(10, 1fr);
      grid-column-end: span 10;
      grid-column-start: 5;
    }
  }
}

.emotion-6 {
  margin-bottom: 1.5rem;
}

.emotion-8 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  margin-top: 0;
  list-style-type: none;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-8 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-8 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

.emotion-8>li {
  position: relative;
}

.emotion-8>li::before {
  top: 0.5rem;
  content: ' ';
  position: absolute;
  border-width: 1rem;
  border: 0.1875rem solid #6E6E73;
  background-color: #6E6E73;
  border-radius: 0;
  left: -1rem;
}

.emotion-10 {
  margin-bottom: 1rem;
}

<div>
  <div
    class="emotion-0 emotion-1"
    dir="ltr"
  >
    <strong
      class="emotion-2 emotion-3"
    >
      At a glance
    </strong>
    <div
      class="emotion-4 emotion-5"
      dir="ltr"
    >
      <div
        class="emotion-6 emotion-7"
      >
        <ul
          class="emotion-8 emotion-9"
          dir="ltr"
          role="list"
        >
          <li
            class="emotion-10 emotion-11"
            role="listitem"
          >
            Level 1
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Gist should render the gist with multiple list items 1`] = `
.emotion-0 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #6E6E73;
  border-top: 0.25rem solid #B80000;
  background-color: #FFFFFF;
  padding-top: 2rem;
  padding-bottom: 2rem;
  padding-left: 1rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-0 ul {
  padding-left: 0.5rem;
}

.emotion-0 ul>li {
  margin-bottom: 10px;
  padding-left: 0.5rem;
}

@media (min-width: 25rem) {
  .emotion-0 ul {
    padding-left: 0;
  }
}

@media (min-width: 63rem) {
  .emotion-0 {
    padding-left: 2rem;
  }

  .emotion-0 ul {
    padding-left: 1rem;
  }

  .emotion-0 ul>li {
    padding-left: 1rem;
  }
}

.emotion-2 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 1.125rem;
  line-height: 1.375rem;
  display: inline-block;
  padding-bottom: 1rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-2 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-2 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (max-width: 14.9375rem) {
  .emotion-4 {
    padding: 0 0.5rem;
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-4 {
    padding: 0 0.5rem;
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-4 {
    padding: 0 1rem;
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-4 {
    padding: 0 1rem;
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-4 {
    margin-left: 20%;
  }
}

@media (min-width: 80rem) {
  .emotion-4 {
    margin-left: 40%;
  }
}

@supports (display: grid) {
  .emotion-4 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      padding: 0 0.5rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      padding: 0 0.5rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-4 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      padding: 0 1rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      padding: 0 1rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-4 {
      grid-template-columns: repeat(10, 1fr);
      grid-column-end: span 10;
      grid-column-start: 5;
    }
  }
}

.emotion-6 {
  margin-bottom: 1.5rem;
}

.emotion-8 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  margin-top: 0;
  list-style-type: none;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-8 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-8 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

.emotion-8>li {
  position: relative;
}

.emotion-8>li::before {
  top: 0.5rem;
  content: ' ';
  position: absolute;
  border-width: 1rem;
  border: 0.1875rem solid #6E6E73;
  background-color: #6E6E73;
  border-radius: 0;
  left: -1rem;
}

.emotion-10 {
  margin-bottom: 1rem;
}

.emotion-14 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: italic;
  font-family: inherit;
  font-weight: inherit;
}

.emotion-18 {
  color: #222222;
  border-bottom: 1px solid #B80000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-18:visited {
  color: #6E6E73;
  border-bottom: 1px solid #6E6E73;
}

.emotion-18:focus,
.emotion-18:hover {
  border-bottom: 2px solid #B80000;
  color: #B80000;
}

<div>
  <div
    class="emotion-0 emotion-1"
    dir="ltr"
  >
    <strong
      class="emotion-2 emotion-3"
    >
      At a glance
    </strong>
    <div
      class="emotion-4 emotion-5"
      dir="ltr"
    >
      <div
        class="emotion-6 emotion-7"
      >
        <ul
          class="emotion-8 emotion-9"
          dir="ltr"
          role="list"
        >
          <li
            class="emotion-10 emotion-11"
            role="listitem"
          >
            Here is a summary of the article
          </li>
          <li
            class="emotion-10 emotion-11"
            role="listitem"
          >
            This component supports 
            <b>
              bold,
            </b>
             
            <i
              class="emotion-14 emotion-15"
            >
              italic,
            </i>
             
            <i
              class="emotion-14 emotion-15"
            >
              <b>
                bold-italic, 
              </b>
            </i>
            <a
              aria-label="links,, external"
              class="emotion-18 emotion-19"
              href="https://www.google.com"
            >
              links,
            </a>
             
            <a
              aria-label="and bold-italic-links,, external"
              class="emotion-18 emotion-19"
              href="https://www.google.com"
            >
              <i
                class="emotion-14 emotion-15"
              >
                <b>
                  and bold-italic-links,
                </b>
              </i>
            </a>
          </li>
          <li
            class="emotion-10 emotion-11"
            role="listitem"
          >
            It can be as long as editorial desires
          </li>
          <li
            class="emotion-10 emotion-11"
            role="listitem"
          >
            It does not support nested lists
          </li>
          <li
            class="emotion-10 emotion-11"
            role="listitem"
          >
            If a sentence is added that is really, really, really long, the text will wrap onto the next line, which is hopefully the behaviour of this sentence right now!
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Gist should render the gist with one list item 1`] = `
.emotion-0 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #6E6E73;
  border-top: 0.25rem solid #B80000;
  background-color: #FFFFFF;
  padding-top: 2rem;
  padding-bottom: 2rem;
  padding-left: 1rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-0 ul {
  padding-left: 0.5rem;
}

.emotion-0 ul>li {
  margin-bottom: 10px;
  padding-left: 0.5rem;
}

@media (min-width: 25rem) {
  .emotion-0 ul {
    padding-left: 0;
  }
}

@media (min-width: 63rem) {
  .emotion-0 {
    padding-left: 2rem;
  }

  .emotion-0 ul {
    padding-left: 1rem;
  }

  .emotion-0 ul>li {
    padding-left: 1rem;
  }
}

.emotion-2 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 1.125rem;
  line-height: 1.375rem;
  display: inline-block;
  padding-bottom: 1rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-2 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-2 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (max-width: 14.9375rem) {
  .emotion-4 {
    padding: 0 0.5rem;
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-4 {
    padding: 0 0.5rem;
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-4 {
    padding: 0 1rem;
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-4 {
    padding: 0 1rem;
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-4 {
    margin-left: 20%;
  }
}

@media (min-width: 80rem) {
  .emotion-4 {
    margin-left: 40%;
  }
}

@supports (display: grid) {
  .emotion-4 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      padding: 0 0.5rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      padding: 0 0.5rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-4 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      padding: 0 1rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      padding: 0 1rem;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-4 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-4 {
      grid-template-columns: repeat(10, 1fr);
      grid-column-end: span 10;
      grid-column-start: 5;
    }
  }
}

.emotion-6 {
  margin-bottom: 1.5rem;
}

.emotion-8 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  margin-top: 0;
  list-style-type: none;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-8 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-8 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

.emotion-8>li {
  position: relative;
}

.emotion-8>li::before {
  top: 0.5rem;
  content: ' ';
  position: absolute;
  border-width: 1rem;
  border: 0.1875rem solid #6E6E73;
  background-color: #6E6E73;
  border-radius: 0;
  left: -1rem;
}

.emotion-10 {
  margin-bottom: 1rem;
}

<div>
  <div
    class="emotion-0 emotion-1"
    dir="ltr"
  >
    <strong
      class="emotion-2 emotion-3"
    >
      At a glance
    </strong>
    <div
      class="emotion-4 emotion-5"
      dir="ltr"
    >
      <div
        class="emotion-6 emotion-7"
      >
        <ul
          class="emotion-8 emotion-9"
          dir="ltr"
          role="list"
        >
          <li
            class="emotion-10 emotion-11"
            role="listitem"
          >
            Level 1
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
